eth/vm/base.py

```diff
@@ -567,37 +562,41 @@ def import_block(self, block: BaseBlock) -> BaseBlock:
         """
         Import the given block to the chain.
         """
-        if self.block.number != block.number:
+        if self.get_block().number != block.number:
             raise ValidationError(
                 "This VM can only import blocks at number #{}, the attempted block was #{}".format(
-                    self.block.number,
+                    self.get_block().number,
                     block.number,
                 )
             )
 
-        self.block = self.block.copy(
-            header=self.configure_header(
-                coinbase=block.header.coinbase,
-                gas_limit=block.header.gas_limit,
-                timestamp=block.header.timestamp,
-                extra_data=block.header.extra_data,
-                mix_hash=block.header.mix_hash,
-                nonce=block.header.nonce,
-                uncles_hash=keccak(rlp.encode(block.uncles)),
-            ),
-            uncles=block.uncles,
+        self.set_block(
```

Comment:
I'm inclined to advocate for removing this method in favor of directly setting via `self._block = ...`

The only place that `set_block` appears to be used is from the tests in `tests/json-fixtures/test_virtual_machine.py` which I'm fine with that accessing the private `vm._block` property for now.

Reply:
this is solved in the 9bd96bb commit